compiler/ilgen/IlBuilder.cpp

```diff
@@ -1222,7 +1222,7 @@ IlBuilder::getOpCode(TR::IlValue *leftValue, TR::IlValue *rightValue)
    TR::ILOpCodes op;
    if (leftValue->getSymbol()->getDataType() == TR::Address)
       {
-      if (leftValue->getSymbol()->getDataType() == TR::Int32)
+      if (rightValue->getSymbol()->getDataType() == TR::Int32)
```

Comment:
i would prefer you fix this problem in the commit where leftValue was originally introduced. Otherwise you introduced a commit with broken code in it that doesn't need to be broken.